src/librbd/internal.cc

```diff
@@ -1998,6 +2024,19 @@ int validate_pool(IoCtx &io_ctx, CephContext *cct) {
     return ictx->get_flags(ictx->snap_id, flags);
   }
 
+  int set_flags(ImageCtx *ictx, uint64_t flags, bool enabled)
```

Comment:
Rename to ```update_flags``` since set_flags implies you are enabling all flags passed as the argument and disabling those not.  Also probably should prevent the API from updating ```RBD_FLAG_OBJECT_MAP_INVALID``` and ```RBD_FLAG_FAST_DIFF_INVALID```.